File: app/src/main/java/com/jakebarnby/filemanager/models/SourceManager.java
package com.jakebarnby.filemanager.models;

import android.util.SparseArray;

import com.jakebarnby.filemanager.activities.source.SourceActivity;
import com.jakebarnby.filemanager.models.files.SourceFile;
import com.jakebarnby.filemanager.util.TreeNode;

/**
 * Created by Jake on 7/29/2017.
 */

public class SourceManager {

    private TreeNode<SourceFile>                    mActiveDirectory;
    private SparseArray<SourceActivity.FileAction>  mCurrentFileActions;

    public SourceManager() {
        this.mCurrentFileActions = new SparseArray<>();
    }

    public TreeNode<SourceFile> getActiveDirectory() {
        return mActiveDirectory;
    }

    public void setActiveDirectory(TreeNode<SourceFile> mActiveDirectory) {
        this.mActiveDirectory = mActiveDirectory;
    }

    public SourceActivity.FileAction getFileAction(int operationId) {
        return mCurrentFileActions.get(operationId);
    }

    public void addFileAction(int operationId, SourceActivity.FileAction action) {
        mCurrentFileActions.put(operationId, action);
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/managers/SourceManager.kt
package com.jakebarnby.filemanager.managers

import android.util.SparseArray
import com.jakebarnby.filemanager.models.FileAction
import com.jakebarnby.filemanager.models.Source
import java.io.Serializable
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by Jake on 7/29/2017.
 */
@Singleton
class SourceManager @Inject constructor() : Serializable {

    val sources: MutableList<Source> = mutableListOf()

    lateinit var activeSource: Source

    private val currentFileActions = SparseArray<FileAction>()

    fun getFileAction(operationId: Int): FileAction? {
        return currentFileActions[operationId]
    }

    fun addFileAction(operationId: Int, action: FileAction) {
        currentFileActions.put(operationId, action)
    }

    fun getSource(id: Int) = sources[id]

}